### PR TITLE
Feature/volume-control

### PR DESCRIPTION
**Pull request by Stanback**
Thanks for making this project. I added support to get and adjust the volume.

I'm using it with homebridge and mqttthing, I've posted my device config [here](https://github.com/tbowmo/chromecastcontrol/pull/14) if anyone is interested

### CODE REVIEW

**Comment by Stanback** [comment]
(FWIW, the section in chromeevent.py, `self.log.warn('!!!!!THIS SHOULD NOT BE CALLED!!!!!')`, does get called when changing the volume.)

**Comment by tbowmo** [comment]
Nice addition.

One thing though, could you change the volume scale from 0-100, instead of 0-1? and do the calculations in python for the correct volume settings on the chromecasts (ie divide by 100 before sendig to the chromecasts, and multiplying by 100 when receiving new status)? I think it would make the interface more intuitive to use, and less conversion has to be done by external sources.

**Comment by tbowmo** [comment]
BTW, the self.log.warn is remains from some testing that I made early on in the project. I'll remove it at some point in time

**Comment by Stanback** [comment]
@tbowmo Sounds good, I updated the code and my example homebridge config with the 0-100 range. I'll do some testing on it this evening.

File: README.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ The following topics will be used:
   * will be used for the currently running app (netflix, spotify, hbo, tidal etc).
 * <MQTT_ROOT>/<FRIENDLY_NAME>/state
   * will be used for the current activity state (playing, paused, buffering)
+* <MQTT_ROOT>/<FRIENDLY_NAME>/volume
+  * will be used for the current volume level (a decimal value between 0 and 1)
```
Comment (tbowmo) [inline README.md]: Not to be nitpicking, but perhaps the documentation should be updated as well? ;) I'll merge once this is updated
```suggestion
  * will be used for the current volume level (an integer value between 0 and 100)
```

**Comment by tbowmo** [comment]
@Stanback have you seen my comment about the documentation?

File: README.md [hunk 2]
```diff
@@ -47,4 +49,6 @@ It's possible to control the chromecasts, by sending a message to the `<MQTT_ROO
 
 if you want to start playing a new media, then you can use the endpoint `<root topic>/cast/control/play`, send the URL for your media as payload
 
+if you want to set the volume, then you can use the endpoint `<MQTT_ROOT>/friendly_name/control/volume`, send a value between 0 and 1 as the payload
```
Comment (tbowmo) [inline README.md]:
```suggestion
if you want to set the volume, then you can use the endpoint `<MQTT_ROOT>/friendly_name/control/volume`, send a value between 0 and 100 as the payload
```

**Review by tbowmo — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Stanback) [reply README.md]: Ah, good catch - will push an update for the docs.

**Comment by Stanback** [comment]
I've committed the requested changes. Sorry about the delay, got busy with other projects and had an issue with avahi that I didn't fix until now.

One relatively minor thing I noticed was that setting the volume works all the time, but, depending on what I'm casting, the volume_level is not always reported correctly. For example, YouTube will report back the correct decimal value, but Soundcloud will always report back 1. I didn't see an obvious solution when inspecting the payload but doesn't seem to be a showstopper.

**Comment by tbowmo** [comment]
Volume is a bit new for me, so haven't used it myself yet, so will have to play a bit with it.. 

Anyways, thanks for the PR.. :) which should be merged now..

**Pull request merged**